src/main.c

```diff
@@ -437,8 +444,40 @@ main(int argc, char** argv)
    memset(&known_fds, 0, sizeof(known_fds));
    memset(message, 0, MISC_LENGTH);
 
+   if (directory_path != NULL)
+   {
+      if (strcmp(directory_path, PGAGROAL_DEFAULT_CONFIGURATION_PATH) == 0)
+      {
+         pgagroal_log_warn("Using the default configuration directory %s, -D can be omitted.", directory_path);
+      }
+      if (access(directory_path, F_OK) != 0)
+      {
+         snprintf(message, MISC_LENGTH, "Configuration directory not found");
+#ifdef HAVE_LINUX
+         sd_notifyf(0, "STATUS=%s: %s", message, directory_path);
+#endif
+         errx(1, "%s: %s", message, directory_path);
+      }
+      if (access(directory_path, R_OK | X_OK) != 0)
+      {
+         snprintf(message, MISC_LENGTH, "Insufficient permissions for directory");
+#ifdef HAVE_LINUX
+         sd_notifyf(0, "STATUS=%s: %s", message, directory_path);
+#endif
+         errx(1, "%s: %s", message, directory_path);
+      }
+
+      pgagroal_check_directory_path(&configuration_path, directory_path, PGAGROAL_DEFAULT_CONF_FILE, PGAGROAL_FALLBACK_CONF_FILE);
```

Comment:
These functions are actually choosing a path and overriding a variable, right? Probably the name of the functions should reflect it (e.g., `pgagroal_normalize_path` or something).